### PR TITLE
[FIX][10.0] add safe_eval context

### PR DESCRIPTION
**Pull request by mourad-ehm**
Add safe_eval context to allow evaluating domain with python params ex(use ref method to get object by xml_id).

### CODE REVIEW

**Comment by mourad-ehm** [comment]
ping @hparfr , @florian-dacosta, @bealdav , @pedrobaeza

File: base_exception/models/base_exception.py [hunk 1]
```diff
@@ -63,11 +63,23 @@ def check_exception_type_consistency(self):
...
+        """ Prepare the context used when evaluating python code
+            :returns: dict -- evaluation context given to safe_eval
+        """
+        return {
+            'uid': self.env.uid,
+            'user': self.env.user,
+            'env': self.env,
+            'self': self,
+        }
+
```
Comment (hparfr) [inline base_exception/models/base_exception.py:73]: I think it should be only `''ref'": self.env.ref` to limit the risk of evading.

**Review by hparfr — APPROVED** [review]
I suggest to limit the scope at a maximum by default (only ref for xml id lookup).
And missing update minor version number.

This feature is really interesting.

File: base_exception/models/base_exception.py (hunk 1 above)
Comment (sbidoul) [reply base_exception/models/base_exception.py:73]: Even `ref` returns a recordset which in turn exposes `sudo`.

Comment (thomaspaulb) [reply base_exception/models/base_exception.py:73]: Does that risk also hold for https://github.com/OCA/server-tools/pull/1606?

**Pull request closed without merging**